### PR TITLE
Initial commit

### PR DESCRIPTION
**Pull request by cdaniluk**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+Creates a SCIM Bridge to enable 1Password SSO w/Okta and other SSO providers. Based on the [1Password SCIM Examples](https://github.com/1Password/scim-examples), but packaged as a ready-to-use module with some security-related improvements. 
+
+<!-- BEGINNING OF PRE-COMMIT-TERRAFORM DOCS HOOK -->
+## Inputs
+
+## Outputs
+
+
+<!-- END OF PRE-COMMIT-TERRAFORM DOCS HOOK -->
```
Comment (sblack4) [inline README.md:37]: Is the pre-commit hook not running?

File: elb.tf [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+resource "aws_security_group_rule" "elb_allow_ingress" {
+  cidr_blocks       = ["0.0.0.0/0"]
+  from_port         = 443
+  protocol          = "tcp"
+  security_group_id = aws_security_group.elb.id
+  to_port           = 443
+  type              = "ingress"
+}
+
+resource "aws_security_group_rule" "elb_allow_this" {
```
Comment (sblack4) [inline elb.tf]:
```suggestion
resource "aws_security_group_rule" "elb_allow_egress" {
```
is that name right?

File: iam.tf [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+    principals {
+      type        = "Service"
+      identifiers = ["ec2.amazonaws.com"]
+    }
+  }
+}
+
+resource "aws_iam_role" "this" {
+  name               = "${var.name}-role"
```
Comment (sblack4) [inline iam.tf:36]: I saw somewhere all these interpolated names were set in the locals. It's really clean especially when you want to allow overrides

File: main.tf [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+}
+
+resource "aws_security_group" "this" {
+  name_prefix = "${var.name}-scim-sg"
+  vpc_id      = var.vpc_id
+
+  tags = merge(
+    var.tags,
+    map(
+      "Name", "${var.name}-scim-sg"
```
Comment (sblack4) [inline main.tf:18]: and if you set the names in locals you get to reference the same local variable in the tags name too

File: variables.tf [hunk 5]
```diff
@@ -0,0 +1,159 @@
+########################################
+# General Vars
+########################################
+
+variable "name" {
+  default     = "1password-scim"
+  description = "Name of this deployment (e.g., prod-1password-scim)"
+  type        = string
+}
```
Comment (sblack4) [inline variables.tf:9]: alphabet:
abcdefghijklmnopqrstuvwxyz

File: variables.tf [hunk 6]
```diff
@@ -0,0 +1,159 @@
...
+
+variable "asg_keypair" {
+  default     = null
+  description = "Optional keypair to associate with instances"
+  type        = string
+}
+
+
```
Comment (sblack4) [inline variables.tf]:
```suggestion
```

File: variables.tf [hunk 7]
```diff
@@ -0,0 +1,159 @@
...
+  type        = string
+}
+
+variable "scim_cache_port" {
+  default     = "6379"
+  description = "Redis cache port (this changes the port SCIM tries to reach redis on but does not change the port redis listens on)"
+  type        = string
+}
+
+
```
Comment (sblack4) [inline variables.tf]:
```suggestion
```

**Review by sblack4 — APPROVED** [review]
I think I understand what's going on here... There's a LB in front of an ASG that does something with 1password?

**Comment by cdaniluk** [comment]
> I think I understand what's going on here... There's a LB in front of an ASG that does something with 1password?

Yeah. They had their own TF but it was a bit overbearing and had some security issues I didn't like.

File: README.md (hunk 1 above)
Comment (cdaniluk) [reply README.md:37]: seemingly not :( I guess that's why it didn't take me 4 attempts to get a clean commit

File: iam.tf (hunk 3 above)
Comment (cdaniluk) [reply iam.tf:36]: example? that sounds neat

Comment (sblack4) [reply iam.tf:36]: It's what I was doing in the elasticache module, then I've been using `coalesce` as a way to allow overides: https://github.com/rhythmictech/terraform-aws-elasticache/blob/4ffd4879ef68a52949fe51fec675e1cb151a36d9/locals.tf#L8

File: README.md (hunk 1 above)
Comment (sblack4) [reply README.md:37]: `pre-commit install && pre-commit run -a`?

File: iam.tf (hunk 3 above)
Comment (cdaniluk) [reply iam.tf:36]: I'mma start doing that more. You'll see it in a PR coming soon.

**Pull request merged**